contracts/Collector.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier:MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interfaces/ICollector.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+contract Collector is ICollector{
+    IERC20 public token;
+    address public owner;
+    RevenuePartner[] private partners;
+
+    constructor(
+        address _owner,
+        IERC20 _token,
+        RevenuePartner[] memory _partners
+    )
+    public
+    validShares(_partners)
+    {   
+        owner = _owner;
+        token = _token;
+        for (uint i = 0; i < _partners.length; i++)
+            partners.push(_partners[i]);
+    }
+
+    function updateShares(RevenuePartner[] memory _partners) 
+    public
+    validShares(_partners)
+    onlyOwner()
+    {
+        uint balance = token.balanceOf(address(this));
+        require(balance == 0, "can't update with balance > 0");
+    
+        delete partners;
+        
+        for (uint i = 0; i < _partners.length; i++)
+            partners.push(_partners[i]);
+    }
+
+    function withdraw() 
+    external 
+    override
+    onlyOwner()
+    {
+        uint balance = token.balanceOf(address(this));
+        require(balance > 0, "no revenue to share");
+
+        for(uint i = 0; i < partners.length; i++)
+            token.transfer(partners[i].beneficiary, SafeMath.div(SafeMath.mul(balance, partners[i].share), 100));
+    }
+
+    function transferOwnership(address _owner)
+    external 
+    override
+    onlyOwner()
+    {
+        owner = _owner;
```

Comment:
I just checked the [Openzeppelin implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/4a9cc8b4918ef3736229a5cc5a310bdc17bf759f/contracts/access/Ownable.sol#L62) and I think we are missing the check of the address received against `address(0)`

Reply:
I know that the owner can specify any address there and we cannot check the validity of the address, but if they specify `address(0)` we are sure the collector will be unusable at all

Reply:
makes sense :+1: